src/Symfony/Component/Validator/Constraint.php

```diff
@@ -197,14 +197,32 @@ protected function normalizeOptions(mixed $options): array
     public function __set(string $option, mixed $value)
     {
         if ('groups' === $option) {
-            $this->groups = (array) $value;
+            $this->groups = $this->normalizeGroups((array) $value);
```

Comment:
I'd rather turn this into `$this->groups = array_map('strval', (array) $value);` and see https://github.com/php/php-src/pull/8825 implemented in PHP 8.2